### PR TITLE
feat: pick somthing from #3609

### PR DESCRIPTION
**Pull request by nuanyang233**
closes #

### CODE REVIEW

File: packages/maskbook/src/database/Persona/Persona.db.ts [hunk 1]
```diff
@@ -64,8 +71,15 @@ const db = createDBAccessWithAsyncUpgrade<PersonaDB, Knowledge>(
                         }
                     }
                 }
+                async function v2_v3() {
+                    db.createObjectStore('relations', { keyPath: ['linked', 'profile'] })
+                    transaction
+                        .objectStore('relations')
+                        .createIndex('linked, profile, favor', ['linked', 'profile', 'favor'], { unique: false })
```
Comment (Jack-Works) [inline packages/maskbook/src/database/Persona/Persona.db.ts:78]: I think it is unique👀

**Review by guanbinrui — DISMISSED** [review]

File: packages/maskbook/src/database/Persona/Persona.db.ts [hunk 2]
```diff
@@ -477,6 +501,86 @@ export async function deleteProfileDB(id: ProfileIdentifier, t: ProfileTransacti
...
+
+    for await (const cursor of t.objectStore('relations').index('linked, profile, favor').iterate()) {
+        if (cursor.value.linked !== linked.toText()) continue
+        if (
+            firstRecord &&
+            options.after &&
+            options.after.linked.toText() !== cursor?.value.linked &&
+            options.after.profile.toText() !== cursor?.value.profile
+        ) {
+            cursor.continue([options.after.linked, options.after.profile, options.after.favor].join(', '))
```
Comment (Jack-Works) [inline packages/maskbook/src/database/Persona/Persona.db.ts:539]: Why did you join the index? I guess the index is an array

Comment (nuanyang233) [reply packages/maskbook/src/database/Persona/Persona.db.ts:539]: yes

Comment (Jack-Works) [reply packages/maskbook/src/database/Persona/Persona.db.ts:539]: So I think the parameter of continue should be an array

**Review by guanbinrui — APPROVED** [review]

**Review by Jack-Works — APPROVED** [review]

**Review by NikoManiac — APPROVED** [review]

**Pull request merged**